styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java

```diff
@@ -46,12 +48,15 @@
 
   private final DockerImageValidator dockerImageValidator;
   private final Duration maybeMaxRunningTimeout;
+  private final Set<String> secretWhitelist;
 
-  private WorkflowValidator(DockerImageValidator dockerImageValidator, Duration maybeMaxRunningTimeout) {
+  private WorkflowValidator(DockerImageValidator dockerImageValidator, Duration maybeMaxRunningTimeout,
+                            Set<String> secretWhitelist) {
     Preconditions.checkArgument(maybeMaxRunningTimeout == null || !maybeMaxRunningTimeout.isNegative(),
         "Max Running timeout should be positive");
-    this.dockerImageValidator = dockerImageValidator;
+    this.dockerImageValidator = Objects.requireNonNull(dockerImageValidator);
```

Comment:
Can you add `requireNonNull` also for `secretWhitelist` to make it consistent and more error proof?

Reply:
Unfortunately this validator is used by CLIMain.java where we can not apply any whitelist check. So `null` means skipping whitelist check.

Reply:
Yes, I see